awsx/ecr/registryImage.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as docker from "@pulumi/docker";
+import * as pulumi from "@pulumi/pulumi";
+import * as schema from "../schema-types";
+import { getDockerCredentials } from "./auth";
+
+export class RegistryImage extends schema.RegistryImage {
+  constructor(
+    name: string,
+    args: schema.RegistryImageArgs,
+    opts?: pulumi.ComponentResourceOptions,
+  ) {
+    super(name, args, opts);
+
+    const creds = pulumi
+      .output(args.repositoryUrl)
+      .apply((url) => getDockerCredentials({ registryUrl: url }, { parent: this }));
+    const provider = new docker.Provider(name, {
```

Comment:
There is no way for the user to customize https://www.pulumi.com/registry/packages/docker/#remote-hosts or anything else about this provider. Possibly OK for now just noting.

Reply:
Good point, I can quickly expose those. I don't think those are gonna be useful for most users (e.g. remote host) as part of pushing a local image, but it doesn't hurt to export them to support future use cases

Reply:
Users can still reach this code with transforms possibly, if really needed.